File: 2021-RemoveAllOccurrencesOfASubstring/2021-RemoveAllOccurrencesOfASubstring.py
# Last updated: 6/11/2026, 11:25:26 AM
class Solution:
    def removeOccurrences(self, s: str, part: str) -> str:
        while(part in s):
            print(s)
            s = s.replace(part, "", 1)
            print(s)
        return s



        
        

File: 2021-RemoveAllOccurrencesOfASubstring/2021-RemoveAllOccurrencesOfASubstring.cpp
// Last updated: 6/11/2026, 11:25:21 AM
class Solution {
public:
    string removeOccurrences(string s, string part) {
        int d;
        while(s.find(part)!=string::npos){
            d = s.find(part);
            s.erase(d, part.length());
        }
        return s;
    }
};